htdocs/adherents/class/adherent_type.class.php

```diff
@@ -662,51 +665,22 @@ public function liste_array($status = -1)
 				while ($i < $nump) {
 					$obj = $this->db->fetch_object($resql);
 
-					$adherenttypes[$obj->rowid] = $langs->trans($obj->label);
+					$adherenttypes_label[$obj->rowid] = $langs->trans($obj->label);
+					$adherenttypes_morphy[$obj->rowid] = $obj->morphy;
+					$adherenttypes_amount[$obj->rowid] = $obj->amount;
+					$adherenttypes_caneditamount[$obj->rowid] = $obj->caneditamount;
 					$i++;
 				}
 			}
 		} else {
 			print $this->db->error();
 		}
-		return $adherenttypes;
-	}
-
-	/**
-	 *  Return the array of all amounts per membership type id
-	 *
-	 *  @param	int		$status			Filter on status of type
-	 *  @return array<int,string>		Array of membership type
-	 */
-	public function amountByType($status = null)
-	{
-		$amountbytype = array();
-
-		$sql = "SELECT rowid, amount";
-		$sql .= " FROM ".MAIN_DB_PREFIX."adherent_type";
-		$sql .= " WHERE entity IN (".getEntity('member_type').")";
-		if ($status !== null) {
-			$sql .= " AND statut = ".((int) $status);
-		}
-
-		$resql = $this->db->query($sql);
-		if ($resql) {
-			$nump = $this->db->num_rows($resql);
-
-			if ($nump) {
-				$i = 0;
-				while ($i < $nump) {
-					$obj = $this->db->fetch_object($resql);
-
-					$amountbytype[$obj->rowid] = $obj->amount;
-					$i++;
-				}
-			}
-		} else {
-			print $this->db->error();
-		}
-
-		return $amountbytype;
+		return array(
```

Comment:
This is a first suggestion you must apply to no break compatibility (the method may be used by external modules).
You add aparameter $mode = 0 as a second parameter of liste_array
When value is 0, the default, the return is the same than currently. No compatibility break or risk.
If value is 1, you return an array.
The array returned should be (the id first and the the property label, ...)
`array(id1 => array('label' => ..., 'morhpy' => ... ), id2 => array('label' => ..., 'morhpy' => ... )`
instead of 
`array('label' => array(id1 => value, id2 => value2), 'morhpy' => array(id1 => valuem, id2 => valuem2)`
This match better dolibarr current dev pratices.

Reply:
Then, I guess I should not delete `public function amountByType ()` neither and instead of modifying 	`public function liste_array($status = -1)` I should create two more functions, one for `morphy` and one for `caneditamount`.
That way, we'll have 4 function to do the same than one but it will assure ther isn't any compatibility break.

What do you think @eldy ?